src/epd_driver/epd_driver.c

```diff
@@ -342,3 +375,80 @@ enum EpdDrawError epd_draw_image(EpdRect area, const uint8_t *data, const EpdWav
     };
     return epd_draw_base(area, data, no_crop, EPD_MODE_DEFAULT, temperature, NULL, waveform);
 }
+
+void epd_set_rotation(enum EpdRotation rotation) {
+    if (rotation<4) {
+      display_rotation = rotation;
+    }
+}
+
+uint8_t epd_get_rotation() {
+    return display_rotation;
+}
+
+uint16_t epd_rotated_display_width() {
+  uint16_t display_width = EPD_WIDTH;
+  switch (display_rotation) {
+          case 1:
+              display_width = EPD_HEIGHT;
+          break;
+          case 3:
+              display_width = EPD_HEIGHT;
+          break;
+      }
+  return display_width;
+}
+
+uint16_t epd_rotated_display_height() {
+  uint16_t display_height = EPD_HEIGHT;
+  switch (display_rotation) {
+          case 1:
+              display_height = EPD_WIDTH;
+          break;
+          case 3:
+              display_height = EPD_WIDTH;
+          break;
+      }
+  return display_height;
+}
+
+uint8_t epd_get_pixel(int x, int y, int fb_width, int fb_height, const uint8_t *framebuffer) {  
+  if (x < 0 || x >= fb_width) {
+   return 0;
+ }
+ if (y < 0 || y >= fb_height) {
+   return 0;
+ }
+ int fb_width_bytes = fb_width / 2 + fb_width % 2;
+ uint8_t buf_val = framebuffer[y * fb_width_bytes + x / 2];
+ if (x % 2) {
+   buf_val = (buf_val & 0xF0) >> 4;
+ } else {
+   buf_val = (buf_val & 0x0F);
+ }
+
+ // epd_draw_pixel needs a 0 -> 255 value
+ return buf_val<<4;
+}
+
+void epd_draw_rotated_image(EpdRect image_area, const uint8_t *image_buffer, uint8_t *framebuffer) {
+  if (epd_get_rotation()!=0) {
```

Comment:
Please use enums instead of the number for the rotation.

Reply:
Good catch. Part in last commit, this last two here: https://github.com/martinberlin/epdiy-rotation/commit/94964b48b1e4450d564ae5d5da1de066a68fa494